test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createReveal({
+      name: name
+    });
+
+    const {revealPeriod} = network.names;
+    for (let i = 0; i < revealPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    {
+      const json = await wallet.createUpdate({
+        name: name,
+        data: {
+          text: ['foobar']
+        }
+      });
+
+      // register directly after reveal
+      const registers = json.outputs.filter(({covenant}) => covenant.type === types.REGISTER);
+      assert.equal(registers.length, 1);
+    }
+
+    // mine a block
+    await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    {
+      const json = await wallet.createUpdate({
+        name: name,
+        data: {
+          text: ['foobar']
+        }
+      });
+
+      // update after register or update
+      const updates = json.outputs.filter(({covenant}) => covenant.type === types.UPDATE);
+      assert.equal(updates.length, 1);
+    }
+  });
+
+  it('should get name resource', async () => {
+    const names = await wallet.getNames();
+    // filter out to names that have data
+    // this test depends on the previous test
+    const [ns] = names.filter(n => n.data.length > 0);
+    assert(ns);
+
+    const state = Resource.decode(Buffer.from(ns.data, 'hex'));
+
+    const resource = await wallet.getResource(ns.name);
+    const res = Resource.fromJSON(resource);
+
+    assert.deepEqual(state, res);
+  });
+
+  it('should create a renewal', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createReveal({
+      name: name
+    });
+
+    const {revealPeriod} = network.names;
+    for (let i = 0; i < revealPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createUpdate({
+      name: name,
+      data: {
+        text: ['foobar']
+      }
+    });
+
+    // mine up to the earliest point in which a renewal
+    // can be submitted, a treeInterval into the future
```

Comment:
Will an attempted `REVEAL` fail if it's too early? Maybe that should be tested here first.